server/controllers/ArticleController.js

```diff
@@ -0,0 +1,252 @@
+            description: article.description,
+            body: article.body,
+            slug: article.slug,
+            image: article.image,
+            createdAt: article.createdAt,
+            updatedAt: article.updatedAt,
+            author: {
+              id: article.User.id,
+              username: article.User.username,
+            },
+          }));
+
+          res.status(200).json({
+            articles: articlesList,
+          });
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'No Articles found',
+          });
+        }
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method updates an article in the database,
+   * It returns a json response of the article updated
+   * It handles the PUT /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static updateArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+
+    Article.find({ where: { slug: req.params.slug } })
+      .then((article) => {
+        if (article) {
+          const updatedArticle = article;
+          updatedArticle.title = title !== undefined ? title : article.title;
+          updatedArticle.description = description
+          !== undefined ? description : article.description;
+          updatedArticle.body = body !== undefined ? body : article.body;
+          updatedArticle.save();
+          User.find({ where: { id: article.authorId } })
+            .then((user) => {
+              res.status(200).json({
+                article: {
+                  id: article.id,
+                  title: article.title,
+                  description: article.description,
+                  body: article.body,
+                  slug: article.slug,
+                  image: article.image,
+                  createdAt: article.createdAt,
+                  updatedAt: article.updatedAt,
+                  author: {
+                    id: user.id,
+                    username: user.username,
+                  },
+                },
+                message: 'Article Updated',
+              });
+            })
+            .catch(next);
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method deletes an article from the database,
+   * It returns a json response of the article deleted
+   * It handles the DELETE /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static deleteArticle(req, res, next) {
+    Article.find({ where: { slug: req.params.slug } })
```

Comment:
Since you are retrieving one `article` I think a `findOne` might be more efficient than a `find` because `findOne` adds a `LIMIT 1` to the query. I might be wrong though

Reply:
resolved